pkg/compute/registration_file.go

```diff
@@ -0,0 +1,40 @@
+package compute
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+)
+
+const (
+	fileMode = 0644
+)
+
+// RegistrationFile is a sentinel on disk whose presence
+// is used to denote that a node has successfully registered
+// with the requester. This file is per-node to allow multiple
+// compute nodes using the same shared directory for config.
+type RegistrationFile struct {
+	path string
+}
+
+func NewRegistrationFile(nodeID string) *RegistrationFile {
+	repo, _ := config.Get[string]("repo")
```

Comment:
we should minimize our usage of global configs https://github.com/bacalhau-project/expanso-planning/issues/474

Better to pass the path instead

Reply:
Done.